x-pack/plugins/security_solution/public/common/components/events_tab/events_query_tab_body.tsx

```diff
@@ -178,3 +192,43 @@ EventsQueryTabBodyComponent.displayName = 'EventsQueryTabBodyComponent';
 export const EventsQueryTabBody = React.memo(EventsQueryTabBodyComponent);
 
 EventsQueryTabBody.displayName = 'EventsQueryTabBody';
+
+const useExternalAlertsInitialUrlState = () => {
+  const replaceUrlParams = useReplaceUrlParams();
+
+  const getInitialUrlParamValue = useGetInitialUrlParamValue<boolean>(EXTERNAL_ALERTS_URL_PARAM);
+
+  const { decodedParam: showExternalAlertsInitialUrlState } = useMemo(
+    () => getInitialUrlParamValue(),
```

Comment:
Made me curious,  and not important, but do you know if destructuring allocates memory for object
Would:
```
const showExternalAlertsInitialUrlState = useMemo(() => getInitialUrlParamState().decodedParam
``` 
Cause a change to memory or memorization?